src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5854,10 +5795,11 @@ private SimpleNameSyntax ParseSimpleName(NameOptions options = NameOptions.None)
             SimpleNameSyntax name = id;
             if (this.CurrentToken.Kind == SyntaxKind.LessThanToken)
             {
-                var pt = this.GetResetPoint();
-                var kind = this.ScanTypeArgumentList(options);
-                this.Reset(ref pt);
-                this.Release(ref pt);
+                ScanTypeArgumentListKind kind;
+                using (var _ = this.GetDisposableResetPoint(resetOnDispose: true))
```

Comment:
This can also be simplified (remove `var _ =`)

Reply:
Done.